networkit/cpp/graph/Graph.cpp

```diff
@@ -919,16 +919,48 @@ std::pair<node, node> Graph::randomEdge(bool uniformDistribution) const {
 	}
 
 	if (uniformDistribution) {
-		return randomEdges(1)[0];
+		/*
+		 * The simple idea here is to interpret all neighborhoods next to each other, resulting
+		 * in a virtual vector of size m. Then we draw a random index and return the edge.
+		 * For undirected edges, the vector has size 2m; but the idea remains. There is one minor
+		 * complication for undirected edges with self-loops: each edge {u,v} with u != v is stored
+		 * twice (once in the neighborhood of u, once in v) but a loop (u, u) is only stored once.
+		 * To equalize the probabilities we reject edges {u,v} with u > v and try again. This leads
+		 * to less than two expected trails in and is only done for undirected graphs with self-loops.
+		 */
+
+		while(true) {
```

Comment:
Space after while?

Reply:
As a side note: in general, I don't think this is important -- such spacing issues do not usually be demanded in reviews.